### PR TITLE
Update the GHA release workflow with trusted publisher enabled

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Enable GHA release by adding trusted publisher.
This workflow will be triggered when a tag is pushed to this repository. The workflow will built the product and publish to Pypi. 
The GitHub release will be created after publishing to Pypi with `artifacts.tar.gz`. 

I will add this workflow as trusted publisher according to this. https://docs.pypi.org/trusted-publishers/adding-a-publisher/

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build-libraries/issues/222

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added pylint `line-too-long` and `invalid-name` ([#590](https://github.com/opensearch-project/opensearch-py/pull/590))
 - Added pylint `pointless-statement` ([#611](https://github.com/opensearch-project/opensearch-py/pull/611))
 - Added a log collection guide ([#579](https://github.com/opensearch-project/opensearch-py/pull/579))
+- - Update the GHA release workflow with trusted publisher enabled  ([#614](https://github.com/opensearch-project/opensearch-py/pull/614))
```
Comment (dblock) [inline CHANGELOG.md:9]: Two `-` here.

Comment (dblock) [inline CHANGELOG.md:9]: I would say "Added GHA release"

**Review by dblock — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (zelinh) [reply CHANGELOG.md]: Updated here. Please help review it again. Thanks.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Btw, we noticed https://github.com/opensearch-project/opensearch-py/issues/609 related to publishing to pypi. Maybe you know how to fix it ...

**Comment by zelinh** [comment]
OIDC on Pypi side for trusted publisher is added.

**Pull request merged**